site/daemon.md

```diff
@@ -74,6 +74,8 @@ fluxd requires setup and offers customization though a multitude of flags.
 |--registry-burst        | `125`      | maximum number of warmer connections to remote and memcache|
 |--registry-insecure-host| []         | registry hosts to use HTTP for (instead of HTTPS) |
 |--docker-config         | `""`       | path to a Docker config file with default image registry credentials |
+|--aws-region            | `""`       | AWS region for authentication, when scanning image in ECR |
+|--aws-registry-id       | `[]`       | ECR registry ID(s) for authentication, when scanning images in ECR (multiple values allowed) |
```

Comment:
Do all registries have to be in the same region as set by `--aws-region` or not necessary? The reason I asked is that above it say "region for authentication", so it sounds like the region has is only for authentication... Would be good to clarify whichever is the case. I am not sure how auth works, I thought that there may be a way to authenticate in one region and call out to other regions, but not too sure.

Reply:
Also, what IDs are these? We should clarify.

Reply:
> Also, what IDs are these? We should clarify.

From looking at `aws ecr help`, (which is not particularly clear either), it sounds like these are actually AWS account IDs.

So I believe if neither `--aws-region` nor `--aws-registry-ids` are set, Flux will simply authenticate agains whatever region and whatever account the cluster is in. Is that the case?
If that is the case, I believe the main use-case for `--aws-registry-ids` is to allow looking at images in multiple accounts (which is common). And `--aws-region` could be used when Flux doesn't run in AWS. Perhaps we could somehow  clarify this in the docs, at least to say that one doesn't need to worry about these flags in a most straight-forward setup.

Reply:
Yes, they are account IDs -- the registry URLs are differentiated by account ID, so the account ID becomes the registry ID. If you reckon it's clearer, I could change this argument name (and perhaps make the connection with registries in the help text).

Reply:
Regarding the region, I see the registry URL _also_ contains a region. So you are right to say that we don't need to be supplied these bits of information -- we could just assume that the necessary credentials are available, and try instigating a session (per region).

These arguments serve a secondary purpose, which is to act as an explicit way of saying "yes, use AWS authentication for these". Perhaps it should be the other way though -- a flag to explicitly switch AWS authentication off, and otherwise assume the ambient credentials are there to be used.

Reply:
> If you reckon it's clearer, I could change this argument name (and perhaps make the connection with registries in the help text).

Yes, I think `--aws-account-ids` seems clearer to me.

> Perhaps it should be the other way though -- a flag to explicitly switch AWS authentication off, and otherwise assume the ambient credentials are there to be used.

I think that would make sense, but I would leave that until a clear use-case for turning it off comes up, unless we already have boolean flags for controlling certain features, in which case this would be a matter of being consistent.